collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -181,6 +179,14 @@ private void handleTaskResult(SaveToDiskResult taskResult, SaveRequest saveReque
         }
     }
 
+    public LiveData<SaveResult> getSavedResult() {
+        return saveResult;
+    }
+
+    public void resumeFormEntry() {
+        saveResult = new MutableLiveData<>(null);
```

Comment:
In my head this needs to be `saveResult.setValue(null)` instead. Otherwise saving multiple times won't work for observations made in `onCreate` of an Activity/Fragment as they'll be observing a different `LiveData` value. Maybe I'm reading this wrong!

We can verify with a test for the `ViewModel`:

```java
@Test
public void resumeFormEntry_clearsSaveResult() {
    LiveData<FormSaveViewModel.SaveResult> saveResult = viewModel.getSavedResult();
    viewModel.saveForm(Uri.parse("file://form"), true, "", false);
    viewModel.resumeFormEntry();
    assertThat(saveResult.getValue(), equalTo(null));
}
```

Reply:
Done.